src/distributions/binomial.rs

```diff
@@ -47,6 +47,35 @@ impl Binomial {
     }
 }
 
+/// Raise a `base` to the power of `exp`, using exponentiation by squaring.
+///
+/// This implementation is based on the one in the `num_traits` crate. It is
+/// slightly modified to accept `u64` exponents.
+fn pow(mut base: f64, mut exp: u64) -> f64 {
+    if exp == 0 {
+        return 1.;
+    }
+
+    while exp & 1 == 0 {
+        base *= base;
+        exp >>= 1;
+    }
+
+    if exp == 1 {
+        return base;
+    }
```

Comment:
This is redundant; is it just a hint? In the original, this path does avoid a "clone", but since we're using `f64` that is irrelevant.

Reply:
You are right, I removed it.